client/maps/action-creators.ts

```diff
@@ -38,10 +40,16 @@ export function uploadLocalMap(path: string, onMapSelect: (map: MapInfoJson) =>
       ({ map }) => {
         onMapSelect(map)
       },
-      () => {
-        externalShowSnackbar(
-          i18n.t('maps.local.uploadMapError', 'An error occurred while uploading the map'),
-        )
+      err => {
+        if (isFetchError(err) && err.code) {
+          if (err.code === FilesErrorCode.MaxFileSizeExceeded) {
+            externalShowSnackbar(i18n.t('maps.local.mapTooLarge', 'The map is too large'))
```

Comment:
"Map is too large" can mean a lot of things other than file size, please be specific with these messages.

I'd prefer both of these errors to be a in a dialog instead of a snackbar as well

Reply:
Do you think we should show the dialog as soon as the upload starts, or just show a dialog when/if the error happens?

If former, should the dialog be a modal one so the user _has_ to wait until the map finishes uploading?

Reply:
I have no preferences